### PR TITLE
LPS-180082, LPS-180083, LPS-180087, LPS-180195, LPS-181943, LPS-181944 | master

### PR DESCRIPTION
**Pull request by andreefarias**
[LPS-180082](https://issues.liferay.com/browse/LPS-180082)
[LPS-180083](https://issues.liferay.com/browse/LPS-180083)
[LPS-180087](https://issues.liferay.com/browse/LPS-180087)
[LPS-180195](https://issues.liferay.com/browse/LPS-180195)
[LPS-181943](https://issues.liferay.com/browse/LPS-181943)
[LPS-181944](https://issues.liferay.com/browse/LPS-181944)

cc: @bronsonsoda, @JerryAdrianoo and @victorhcunha 

Related this PRs: #692, #721 and #722

### CODE REVIEW

**Comment by andreefarias** [comment]
Thanks for the job, @bronsonsoda, @JerryAdrianoo and @victorhcunha! I made some small adjustments on the [new commit](https://github.com/liferay-bpm-qa/liferay-portal/pull/753/commits/c73c8e01ce579ab311c2fea2925883df93f8c808).

LGTM! 🎉

**Comment by andreefarias** [comment]
ci:test:sf

**Comment by timpak** [comment]
Taking a look, this might take a little while to review

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 1]
```diff
@@ -106,50 +106,83 @@ definition {
...
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Simple T-Shirt");
+
+			FrontStore.assertProductDetails(
+				productFullDescription = "Simple T-Shirt Full Description",
+				productName = "Simple T-Shirt",
+				productShortDescription = "Simple T-Shirt Short Description");
+		}
+
+		task ("And given an action to add a Commerce Product Group entry after deleting a Commerce Product entry") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Hey @andreefarias I think something went wrong while cherry picking the commit. Can you fix this by cherry picking https://github.com/liferay-bpm-qa/liferay-portal/pull/721/commits/3a1d8d823fd27b9d42f185896beb3e196f76967a again?

Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: From reviewing the test from the original commit, the test is fine but we should remove the extra assertion to check the status of the action.

Comment (timpak) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: We also shouldn't be navigating to the minium site to assert the entry was created in the given. There's another example of this in the next test.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 2]
```diff
@@ -3711,4 +3744,93 @@ definition {
...
+			ApplicationsMenu.gotoSite(site = "Minium");
+
+			CommerceAcceleratorsInitializer.viewMiniumCardName(productName = "Simple T-Shirt");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Simple T-Shirt");
+
+			FrontStore.assertProductDetails(
+				productFullDescription = "Simple T-Shirt Full Description",
+				productName = "Simple T-Shirt",
+				productShortDescription = "Simple T-Shirt Short Description");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: This is outside of the scope of the test. We shouldn't be asserting the product entry was created since that's not the purpose of the test.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 3]
```diff
@@ -3711,4 +3744,93 @@ definition {
...
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Commerce Product");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.assertObjectActions(
+				actionActive = "Yes",
+				actionLabel = "Custom Action",
+				status = "Success");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: This assertion is outside of the scope of the test. We don't need to check the status of the action here since that's not the purpose of the test. Our base action test should be testing this flow.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 4]
```diff
@@ -3509,6 +3509,95 @@ definition {
...
+			ApplicationsMenu.gotoSite(site = "Minium");
+
+			CommerceAcceleratorsInitializer.viewMiniumCardName(productName = "Simple T-Shirt");
+
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "Simple T-Shirt");
+
+			FrontStore.assertProductDetails(
+				productFullDescription = "Simple T-Shirt Full Description",
+				productName = "Simple T-Shirt",
+				productShortDescription = "Simple T-Shirt Short Description");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Same thing here, no need to include this assertion since it's outside of the scope of the test.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 5]
```diff
@@ -3509,6 +3509,95 @@ definition {
...
+
+			ObjectPortlet.selectSystemObject(label = "Commerce Product");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.assertObjectActions(
+				actionActive = "Yes",
+				actionLabel = "Custom Action",
+				status = "Success");
+		}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Same thing here, we don't need to make this assertion since this isn't within the scope of the test.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 6]
```diff
@@ -3319,6 +3319,65 @@ definition {
...
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 180195");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.assertObjectActions(
+				actionActive = "Yes",
+				actionLabel = "Custom Action",
+				status = "Success");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: We should be asserting the second account entry is created instead of assertion the status of the action. The main purpose of the test is to assert the account entry is created after all.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 7]
```diff
@@ -106,6 +106,128 @@ definition {
...
+		}
+	}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: This looks like a sorting issue, please make the fix in the original commit.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 8]
```diff
@@ -106,6 +106,128 @@ definition {
...
+
+	@description = "LPS-173537 - Verify creating a custom object entry triggers an action to update an Account entry"
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:261]: I made a mistake, the description should be:

`Verify updating a custom object entry triggers an action to add an Account entry`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 9]
```diff
@@ -106,6 +106,128 @@ definition {
...
+
+			ObjectAdmin.goToActionDetails(viewEntry = "Entry Test");
+
+			ObjectAdmin.editEntryName(newEntryValue = "Entry Test Edited");
+		}
+
+		task ("Then the action is triggered and an Account entry is added") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject180083");
+
+			ObjectPortlet.viewEntry(entry = "Entry Test Edited");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: We don't need to include this assertion since it's outside of the scope of the test. We have other tests that test editing entries so we don't need to duplicate this assertion here.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 10]
```diff
@@ -165,6 +165,69 @@ definition {
...
+
+			ObjectPortlet.deleteEntry(entry = "Entry Test");
+		}
+
+		task ("Then the action is triggered and an Account entry is added") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject180087");
+
+			AssertElementNotPresent(
+				locator1 = "ObjectAdmin#ANY_TAB_ON_ENTRY",
+				value1 = "Entry Test");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Same thing here, we don't need to assert the object entry is deleted since we already have tests that assert this.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 11]
```diff
@@ -106,50 +106,83 @@ definition {
...
 
-	@description = "LPS-180070 - Verify creating a Commerce Product entry triggers an action to add a user"
+	@description = "LPS-181943 - Verify deleting a Commerce Product entry triggers an action to add a Commerce Product Group entry"
 	@priority = 5
-	test CanAddUserAfterCreatingCommerceProductEntry {
+	test CanAddCommerceProductGroupEntryAfterDeletingCommerceProductEntry {
 		property portal.acceptance = "true";
 
-		task ("Given a Commerce Product object") {
+		task ("Given a minium site and Commerce Product object") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Given a minium site and Commerce Product entry

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 12]
```diff
@@ -3509,6 +3509,95 @@ definition {
 			actionLabel = "Update Action Label");
 	}
 
+	@description = "LPS-181944 - Verify updating a Commerce Product entry triggers an action to update the Commerce Product entry"
+	@priority = 5
+	test CanUpdateCommerceProductEntryAfterUpdatingCommerceProductEntry {
+		property portal.acceptance = "true";
+
+		task ("Given a minium site and Commerce Product") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Given a minium site and Commerce Product entry

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 13]
```diff
@@ -165,6 +165,63 @@ definition {
...
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "true",
+				objectName = "CustomObject180082");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject180082");
+		}
+
+		task ("And given an action to create a new Account entry after adding an entry on Custom Object") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: `And given an action to add an Account entry after creating a custom object entry`

@andreefarias please make sure people are using the correct text for the tests. We absolutely cannot allow people to use text that wasn't approved by me.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 14]
```diff
@@ -165,6 +165,63 @@ definition {
...
+				fieldName = "customObjectField",
+				objectName = "CustomObject180082",
+				value = "Entry Test");
+		}
+
+		task ("Then the action is triggered and an Account entry is added") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject180082");
+
+			ObjectPortlet.viewEntry(entry = "Entry Test");
+
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: No need to assert this since it isn't within the scope of the test. We only need to assert the account entry is created.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 15]
```diff
@@ -111,25 +111,10 @@ definition {
...
-				fieldBusinessType = "Integer",
-				fieldLabelName = "Custom Field",
-				fieldName = "customField",
-				fieldType = "Integer",
-				isRequired = "false",
-				objectName = "CustomObject180195");
-
-			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject180195");
-
+		task ("Given an Account object and given an action to add a new Account entry after creating an Account entry") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: I have no idea what https://github.com/liferay-bpm-qa/liferay-portal/pull/753/commits/c73c8e01ce579ab311c2fea2925883df93f8c808 is doing, I can't review this commit since the changes are really hard to follow. It's better to push back code to the original author for them to make changes instead of making additional commits to fix their mistakes.

If you want to fix their mistakes, then you should do it this way:

1. Cherry pick first commit
2. Fix mistakes from the first commit in a new commit with a clear commit message explaining what you're fixing and why
3. Cherry pick second commit
4. Fix mistakes from the second commit in a new commit with a clear commit message explaining what you're fixing and why

This makes it a lot easier to understand what your changes are doing and makes it easier to review.

**Review by timpak — CHANGES_REQUESTED** [review]
@andreefarias can you take a look at my comments below? You need to cherry pick the commits cleanly and fix any mistakes one at a time with each new commit you cherry pick. Fixing everything at the end makes it really hard to review the changes. 

In general, I don't recommend merging commits like this from different authors if they haven't passed review since it's hard to give them direct feedback.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase (hunk 1 above)
Comment (andreefarias) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase]: Sure

**Comment by andreefarias** [comment]
I'll check your comments and return the PRs for our authors with the requests necessary. Thanks @timpak!

**Comment by andreefarias** [comment]
I'll return the ticket to the authors and close this PR.
